### ISSUE TITLE
Can't pass tests on the master branch.

### ISSUE THREAD

**Issue by cryptcoin-junkey** [rust-bitcoin/rust-bitcoin#328]
For reproductive, I tried to run `cargo test` on Docker.

```Dockerfile
FROM rust:1.22

RUN git clone https://github.com/rust-bitcoin/rust-bitcoin.git
WORKDIR rust-bitcoin
RUN cargo build
RUN cargo test
```

<details>
  <summary>Logs on `docker build .`</summary>

```
Sending build context to Docker daemon  2.048kB
Step 1/5 : FROM rust:1.22
 ---> 0a08c365b9ac
Step 2/5 : RUN git clone https://github.com/rust-bitcoin/rust-bitcoin.git
 ---> Using cache
 ---> ed0494dd91a5
Step 3/5 : WORKDIR rust-bitcoin
 ---> Using cache
 ---> 6c96cc401b7b
Step 4/5 : RUN cargo build
 ---> Using cache
 ---> d94872795d9f
Step 5/5 : RUN cargo test
 ---> Running in 6d17d84f7926
 Downloading serde_json v1.0.40
 Downloading serde_test v1.0.100
 Downloading serde_derive v1.0.98
 Downloading itoa v0.4.4
 Downloading ryu v1.0.0
 Downloading syn v0.15.44
 Downloading proc-macro2 v0.4.30
 Downloading quote v0.6.13
 Downloading unicode-xid v0.1.0
   Compiling itoa v0.4.4
   Compiling syn v0.15.44
   Compiling unicode-xid v0.1.0
   Compiling proc-macro2 v0.4.30
   Compiling ryu v1.0.0
   Compiling serde_test v1.0.100
   Compiling serde_json v1.0.40
   Compiling quote v0.6.13
   Compiling serde_derive v1.0.98
   Compiling bitcoin v0.20.0 (file:///rust-bitcoin)
error[E0432]: unresolved import `secp256k1::rand::thread_rng`
   --> src/util/contracthash.rs:287:9
    |
287 |     use secp256k1::rand::thread_rng;
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^ no `thread_rng` in `rand`

error[E0425]: cannot find function `thread_rng` in this scope
   --> src/util/merkleblock.rs:509:23
    |
509 |         let mut rng = thread_rng();
    |                       ^^^^^^^^^^ not found in this scope
    |
help: possible candidate is found in another module, you can import it into scope
    |
496 |     use std::__rand::thread_rng;
    |

error[E0412]: cannot find type `ThreadRng` in this scope
   --> src/util/merkleblock.rs:696:40
    |
696 |         fn damage(&mut self, rng: &mut ThreadRng) {
    |                                        ^^^^^^^^^ not found in this scope
    |
help: possible candidate is found in another module, you can import it into scope
    |
496 |     use std::__rand::ThreadRng;
    |

error: unused import: `secp256k1::rand::prelude`
   --> src/util/merkleblock.rs:500:9
    |
500 |     use secp256k1::rand::prelude::*;
    |         ^^^^^^^^^^^^^^^^^^^^^^^^
    |
note: lint level defined here
   --> src/lib.rs:44:9
    |
44  | #![deny(unused_imports)]
    |         ^^^^^^^^^^^^^^

error: aborting due to 4 previous errors

error: Could not compile `bitcoin`.

To learn more, run the command again with --verbose.
The command '/bin/sh -c cargo test' returned a non-zero code: 101
```
</details>

**Comment by cryptcoin-junkey** [comment]
I'm not sure as I'm new to Rust. But this was caused by API breakages in `rand` ?

**Comment by apoelstra** [comment]
No, I think it's https://github.com/rust-bitcoin/rust-secp256k1/issues/157 (API breakage in rust-secp).

**Comment by elichai** [comment]
:(

**Comment by apoelstra** [comment]
Should be fixed now with rust-secp 0.15.5

**Comment by cryptcoin-junkey** [comment]
@apoelstra Thanks for your kind support.
I checked to build with same Dockerfile above.
binaries were built with rust-secp 0.15.5 and all tests was passed.
So I close this issue as resolved.
